### ISSUE TITLE
unexpected failure type for check50.run("X")..stdout("lala") if X returns empty string

### ISSUE THREAD

**Issue by pazz** [cs50/check50#256]
I have a check that calls run and stdout like this:

```python
def mas_check_stuff():
    check50_java.run("some command").stdout("expected)
```

and check50 produces a json report like this:

```json
        {
            "name": "mas_check_stuff",
            "description": "bla",
            "passed": false,
            "log": [
                "running some command",
            ],
            "cause": {
                "rationale": "Did not find \"expected\"",
                "help": null,
                "missing_item": "expected",
                "collection": ""
            },
            "data": {},

        },
```

Here, the student's submission actually produced the empty string.
It looks like check50.run().stdout() produces a `check50.Missing` exception. But that is not documented like this!
I am expecting a `Mismatch` for further processing..

https://cs50.readthedocs.io/projects/check50/en/latest/api/#check50.run.stdout

**Comment by pazz** [comment]
Sorry: the command `X` does not produce the empty string but it does not terminate. Still, this should raise something that is documented. Thanks!

**Comment by ivanharvard** [comment]
Just patched the docs to reflect this! Thanks for the help!
